csharp/extractor/Semmle.Extraction/Entities/Base/CachedEntity`1.cs

```diff
@@ -30,6 +31,7 @@ protected CachedEntity(Context context) : base(context)
     /// <typeparam name="TSymbol">The type of the symbol.</typeparam>
     public abstract class CachedEntity<TSymbol> : CachedEntity where TSymbol : notnull
     {
+        [NotNull]
```

Comment:
There are a couple of `#nullable disable warnings` scattered around the code that relate to cached entities, such as in `Compilation`, `Type`, `VarargsParam`. It might be worth checking if these preprocessor directives can be also removed, or if any of the classes benefit from the `Symbol` not being marked as not null.

Reply:
Oh no! Good that you bring this up! Then the change as it is might not work! I will look into it.

Reply:
I just took a look at the inheritance hierarchy for `CacheEntity`.
There are many places where "null" is passed specifically as the `Symbol` and then most likely all the methods that rely on `Symbol` are overridden.
If it should really be fixed, then we probably need to distinguish between cached entities and nullable cached entities, which requires a larger re-write.

I think I will re-introduce the null check in the `GetHashCode` method to make sure that nothing gets broken (but in principle this method should always be overwritten in case Symbol could be null).

None of the disable nullable warnings can be removed as the Symbol can be (or is actually) null.
However, I think it is still a good idea to tag the property as "not null" as all the "correct" usages of the class should suffer from the hacks where the warning is disabled.